azerothcore/azerothcore-wotlk
# fix(Core/ArenaTeam): Arena team queue protections

## CHANGES PROPOSED:
- If an arena team member is removed from the team, all other members are forced to leave the current arena queue.
- Arena team members cannot be removed from the team while invited to an arena match.
- Arena teams cannot be disbanded while a team member is in an arena queue
- Arena teams cannot be disbanded while a team member is invited to an arena match.

I take no credit for this. Just porting @dcombat's [commit](https://github.com/dcombat/azerothcore-wotlk/commit/d034186b4528d9d30c531ac01a525e594da1503a) from [here](https://github.com/azerothcore/azerothcore-wotlk/issues/2720#issuecomment-596447612)

## ISSUES ADDRESSED:
Closes https://github.com/azerothcore/azerothcore-wotlk/issues/2720


## HOW TO TEST THE CHANGES:

Create an arena team and join an arena queue

<details><summary>While in arena queue</summary>

1. Remove a team member

2. Attempt to disband the team

</details>

<details><summary>While invited to arena match</summary>

1. Attempt to remove a team member

2. Attempt to disband the team
</details>

**Edit by Shin:** also try to play with Arenas in general and make sure nothing is broken (queue, teams, scores, etc..).

## Target branch(es):


## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
